src/type.h

```diff
@@ -1169,9 +1141,16 @@ class FunctionType : public Type {
         etc. */
     const llvm::SmallVector<SourcePos, 8> paramPositions;
 
+    mutable unsigned int flags;
+
     mutable const FunctionType *asMaskedType, *asUnmaskedType;
 
-    const SourcePos pos;
+    FunctionType(const FunctionType &other);
+
+    const FunctionType *Clone() const override;
+    const FunctionType *CloneWithRetTypeAndNewParamTypes(const Type *newReturnType,
```

Comment:
Maybe rename it to `CloneWithSignature` but it's up to you.

Reply:
Done.